File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int DeepestLeavesSum(TreeNode root) {

        Queue<TreeNode> queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        TreeNode curr;
        int sum = 0;

        while(queue.Count != 0){

            int length = queue.Count;
            sum = 0;
            for(int i = 0; i < length; i++){
                curr = queue.Dequeue();
                Console.WriteLine(sum);
                if(curr.right==null && curr.left==null) sum+=curr.val;

                if(curr.left != null) queue.Enqueue(curr.left);
                if(curr.right != null) queue.Enqueue(curr.right);
                
                    
            }


        }
        return sum;

        
        
    }
}

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque

class Solution:
    def deepestLeavesSum(self, root: Optional[TreeNode]) -> int:

        if not root:
            return 0
        return self.bfs(root)
        
    
    def bfs(self, node):

        
        q = deque([node])


        while q:
            sumOfLevel = 0

            for i in range(len(q)):
                
                currNode = q.popleft()
                if currNode.left:
                    q.append(currNode.left)
                if currNode.right:
                    q.append(currNode.right)
                
                sumOfLevel += currNode.val

        return sumOfLevel
            
            


    









            

        